src/main/java/nextstep/interfaces/interceptor/LoginInterceptor.java

```diff
@@ -0,0 +1,67 @@
+package nextstep.interfaces.interceptor;
+
+import lombok.RequiredArgsConstructor;
+import nextstep.domain.model.template.enumeration.Role;
+import nextstep.domain.model.template.annotation.NoAuth;
+import nextstep.infra.jwt.AuthorizationExtractor;
+import nextstep.infra.jwt.JwtTokenProvider;
+import nextstep.infra.exception.auth.AuthorizationException;
+import nextstep.infra.exception.auth.NoAccessAuthorityException;
+import nextstep.infra.exception.auth.NoSuchTokenException;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@RequiredArgsConstructor
+public class LoginInterceptor implements HandlerInterceptor {
+    private static final String ACCESS_TOKEN = "accessToken";
+    private static final String TOKEN_TYPE = "tokenType";
+    private static final String BEARER_TYPE = "Bearer";
+
+    private final JwtTokenProvider jwtTokenProvider;
+    private final Role role;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        if (hasNoAuth(handler)) {
+            return true;
+        }
+
+        String token = AuthorizationExtractor.extract(request);
+
+        request.setAttribute(ACCESS_TOKEN, token);
+        request.setAttribute(TOKEN_TYPE, BEARER_TYPE);
+
+        isTokenExists(token);
+        isTokenValid(token);
+        isRoleValid(token);
```

Comment:
유효성 검사가 끝난 뒤에 셋팅해도 괜찮을거 같아요~! 

```suggestion
        isTokenExists(token);
        isTokenValid(token);
        isRoleValid(token);
        
        request.setAttribute(ACCESS_TOKEN, token);
        request.setAttribute(TOKEN_TYPE, BEARER_TYPE);
```

Reply:
오 넵 그쪽이 훨씬 나을 것 같아요. 감사합니다!